rest/route/host_agent.go

```diff
@@ -419,12 +419,12 @@ func assignNextAvailableTask(ctx context.Context, env evergreen.Environment, tas
 		}
 		if err != nil {
 			if !errors.Is(err, context.Canceled) && ctx.Err() == nil {
```

Comment:
Is it fine that this line needs to check for context cancelled but none of the others?

Reply:
Taking a quick look, I don't think the others do- but they might. I think we should add it as needed since the context cancelling might just be an operation taking long or a request being cancelled